apollo-router/src/error.rs

```diff
@@ -423,7 +423,7 @@ struct ParserError {
 impl ParseErrors {
     #[allow(clippy::needless_return)]
     pub(crate) fn print(&self) {
-        if LevelFilter::current() == LevelFilter::OFF {
+        if LevelFilter::current() == LevelFilter::OFF && cfg!(not(debug_assertions)) {
             return;
```

Comment:
This hack is to make tests take the potentially-panicky path

Reply:
could you explain a bit more this change and the next on line 436? The rest of the PR looks nice but that part feels weird

Reply:
There was a panicky bug. I want first to reproduce the panic in a test, then fix it. There were two obstacles to reproduction:

* `LevelFilter::current()` appears to be OFF while running test, so the panicky code path was not taken at all. This change causes some logging to stdout to happen even if logging is nominally disabled, when compiled with debug assertions such as for testing. In the test harness however, stdout is captured by default and only shown when a test fails. So this doesn’t normally add visible output.

  I don’t really like this weird condition, but it’ll goo away when we drop AST-based parsing. This shouldn’t print to stdout in the first place. This PR fixes the panic (and tests the fix) without waiting for that larger refactor.

* The second obstacle is https://github.com/rust-lang/rust/issues/107118 which would swallow the panic when stdout is captured by the test harness. As commented, adding `format!` works around that.